lib/phoenix_live_view/tag_engine.ex

```diff
@@ -1160,6 +1163,93 @@ defmodule Phoenix.LiveView.TagEngine do
     end
   end
 
+  defp handle_keyed_comprehension(key_expr, state, tag_meta) do
+    # for now, we only support plain tags because we don't know if a function component
+    # renders to a single tag which is required by live components
+    case state.tag_handler.classify_type(tag_meta.tag_name) do
+      {:tag, _} ->
+        :ok
+
+      other ->
+        type =
+          case other do
+            {:local_component, _} -> "function components"
+            {:remote_component, _} -> "function components"
+            # special attrs for slots are handled separately, see wrap_special_slot
+            other -> inspect(other)
+          end
+
+        raise_syntax_error!(
+          "keyed :for comprehensions are not supported on #{type}",
+          tag_meta,
+          state
+        )
+    end
+
+    for_expr =
+      tag_meta[:for] || raise_syntax_error!("cannot use :key without :for", tag_meta, state)
+
+    if_expr = tag_meta[:if]
+    for_assigns = extract_variables(for_expr)
+    vars_changed = Enum.map(for_assigns, fn {name, _} -> name end)
+
+    ast =
+      invoke_subengine(state, :handle_body, [
+        [skip_require: true, root: true, vars_changed: vars_changed]
+      ])
+
+    vars_changed = Macro.var(:vars_changed, Phoenix.LiveView.Engine)
+
+    new_ast =
+      quote do
+        Phoenix.LiveView.TagEngine.component(
+          &Phoenix.Component.live_component/1,
+          %{
+            id: {unquote(state.caller.module), unquote(tag_meta.line), unquote(key_expr)},
```

Comment:
```suggestion
            id: {unquote(state.caller.module), unquote(tag_meta.line), unquote(tag_meta.column), unquote(key_expr)},
```

so we are 100% fool proof?